src/Credentials/CredentialProvider.php

```diff
@@ -54,14 +54,16 @@ class CredentialProvider
     const ENV_SHARED_CREDENTIALS_FILE = 'AWS_SHARED_CREDENTIALS_FILE';
 
     /**
-     * Create a default credential provider that first checks for environment
-     * variables, then checks for the "default" profile in ~/.aws/credentials,
+     * Create a default credential provider that
+     * first checks for environment variables,
+     * then check for role-based auth in ~/.aws/config,
+     * then check for credential_process in the "default" profile in ~/.aws/credentials,
+     * then checks for the "default" profile in ~/.aws/credentials,
+     * then for credential_process in the "default profile" profile in ~/.aws/config,
```

Comment:
I believe this is referring to the SSO sign on, which should be before the first credential_process

Reply:
Yup, it does and it is first in there. All good.